validation.lua

```diff
@@ -118,5 +124,12 @@ create_token = function (self, purpose, username, email)
         email,
         mail_subjects[purpose] .. username,
         mail_bodies[purpose],
-        self:build_url('/users/' .. username .. '/' .. purpose .. '/' .. token_value))
+        self:build_url(self:url_for(
+            purpose,
+            {
+                username = url.build_path({username}),
```

Comment:
For some reason this function doesn't encode `=` and `&`, which I think makes this a problem for query strings, but maybe there's some happy medium?

I guess I need to do more research.